From af2f24e9869059ef0328f7fc77d8c055900d5062 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Sat, 12 Dec 2020 10:46:06 +0100
Subject: [PATCH 1/2] meta: image: Account for Ubuntu differences in
 do_copy_boot_files

Ubuntu places kernel and initrd links under /boot. Furthermore, it makes
the kernel unreadable for non-root users. Account for the latter by
cat'ing the kernel under sudo, redirecting the output to the deployment
artifact so that it is owned by the building user.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 meta/classes/image.bbclass | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/meta/classes/image.bbclass b/meta/classes/image.bbclass
index 74fc8500..eddc4449 100644
--- a/meta/classes/image.bbclass
+++ b/meta/classes/image.bbclass
@@ -132,15 +132,18 @@ EOF
 
 do_copy_boot_files[dirs] = "${DEPLOY_DIR_IMAGE}"
 do_copy_boot_files() {
-    kernel="$(realpath -q '${IMAGE_ROOTFS}/vmlinuz')"
+    kernel="$(realpath -q '${IMAGE_ROOTFS}'/vmlinu[xz])"
     if [ ! -f "$kernel" ]; then
-        kernel="$(realpath -q '${IMAGE_ROOTFS}/vmlinux')"
+        kernel="$(realpath -q '${IMAGE_ROOTFS}'/boot/vmlinu[xz])"
     fi
     if [ -f "$kernel" ]; then
-        cp -f "$kernel" '${DEPLOY_DIR_IMAGE}/${KERNEL_IMAGE}'
+        sudo cat "$kernel" > "${DEPLOY_DIR_IMAGE}/${KERNEL_IMAGE}"
     fi
 
     initrd="$(realpath -q '${IMAGE_ROOTFS}/initrd.img')"
+    if [ ! -f "$initrd" ]; then
+        initrd="$(realpath -q '${IMAGE_ROOTFS}/boot/initrd.img')"
+    fi
     if [ -f "$initrd" ]; then
         cp -f "$initrd" '${DEPLOY_DIR_IMAGE}/${INITRD_IMAGE}'
     fi
-- 
2.26.2

